neuralop/layers/codano.py

```diff
@@ -0,0 +1,460 @@
+from functools import partial
+import logging
+import numpy as np
+import torch
+from einops import rearrange
+import torch
+from torch import nn
+import torch.nn.functional as F
+from .fno_block import FNOBlocks
+from .spectral_convolution import SpectralConv1d, SpectralConv2d, SpectralConv3d
+
+
+AffineNormalizer2D = partial(nn.InstanceNorm2d, affine=True)
+AffineNormalizer3D = partial(nn.InstanceNorm3d, affine=True)
+
+# Codomain Attention Blocks
+
+class CODABlocks(nn.Module):
```

Comment:
The overall structure looks great, thanks Ashiq! But can you add some more comments as well? and remove the self.logger statements? we dont usually add that in our layers.

Reply:
ok.